### PR TITLE
Multi python

### PR DESCRIPTION
**Pull request by olliethomas**

### CODE REVIEW

File: .github/workflows/mypy_check.yml [hunk 1]
```diff
@@ -3,24 +3,42 @@ name: Mypy
 on:
   pull_request:
     branches:
-    - master
+    - multi-python
```
Comment (olliethomas) [inline .github/workflows/mypy_check.yml:6]: eurgh - missed this

Comment (olliethomas) [inline .github/workflows/mypy_check.yml:6]:
```suggestion
    - master
```

File: setup.py [hunk 2]
```diff
@@ -56,7 +56,9 @@
     ],
     python_requires=">=3.6",
     # use `pip install data-science-types[dev]` to install development packages
-    extras_require={"dev": ["black", "mypy", "flake8", "flake8-pyi"]},
+    extras_require={
```
Comment (olliethomas) [inline setup.py:59]: btw, did a new environment to run the 36 tests locally. Couldn't run without installing these extra packages for the tests, so added them

File: check_all.sh [hunk 3]
```diff
@@ -5,6 +5,6 @@ echo "Begin check..." \
 && python gen_pyi.py \
 && black -l 100 -t py36 --pyi --fast matplotlib-stubs/pyplot.pyi.in \
 && flake8 *-stubs \
-&& python -m pytest -vv tests/ \
-&& mypy tests \
+&& python -m pytest -vv tests/py37+ \
```
Comment (olliethomas) [inline check_all.sh:8]: btw, this is where I was thinking of using the `python -V` to have a conditional about which tests to run/check based on the python version.

**Comment by olliethomas** [comment]
Looking at the checks - this needs the branch protection rules updating

**Comment by thomkeh** [comment]
I think you should remove the tests for python 3.7+ and use only the ones for python 3.6. They will still work with 3.7+ and I don't like maintaining two almost identical sets of tests.

Yeah and then we'll need to change the protection rules, but let's only do that once #122 has been merged.

File: .github/workflows/test_correctness.yml [hunk 4]
```diff
@@ -3,23 +3,40 @@ name: Pytest
 on:
   pull_request:
     branches:
-    - master
+    - multi-python
```
Comment (thomkeh) [inline .github/workflows/test_correctness.yml:6]:
```suggestion
    - master
```

**Comment by thomkeh** [comment]
closed in favor of #123

**Pull request closed without merging**